File: www/NowPlaying.js
//
// NowPlaying.js
// Now Playing Cordova Plugin
//
// Created by François LASSERRE on 30/08/15.
// Copyright 2015 François LASSERRE. All rights reserved.
// MIT Licensed
//
var exec = require('cordova/exec');

function NowPlaying(){};

NowPlaying.prototype.updateMetas = function(artist, title, album, cover, success, error) {
    exec(success, error, "NowPlaying", "updateMetas", [artist, title, album, cover]);
};

module.exports = new NowPlaying();

File: src/ios/NowPlaying.m
//
// NowPlaying.m
// Now Playing Cordova Plugin
//
// Created by François LASSERRE on 30/08/15.
// Copyright 2015 François LASSERRE. All rights reserved.
// MIT Licensed
//

#import "NowPlaying.h"

@implementation NowPlaying

- (void)updateMetas:(CDVInvokedUrlCommand*)command
{
    NSString *artist = [command.arguments objectAtIndex:0];
    NSString *title = [command.arguments objectAtIndex:1];
    NSString *album = [command.arguments objectAtIndex:2];

    if (NSClassFromString(@"MPNowPlayingInfoCenter"))  {
        MPNowPlayingInfoCenter *infoCenter = [MPNowPlayingInfoCenter defaultCenter];
        infoCenter.nowPlayingInfo = [NSDictionary dictionaryWithObjectsAndKeys:
                                     artist, MPMediaItemPropertyArtist,
                                     title, MPMediaItemPropertyTitle,
                                     album, MPMediaItemPropertyAlbumTitle,
                                     nil];
    }
}

@end